cdap-docs/reference-manual/source/http-restful-api/security.rst

```diff
@@ -206,26 +215,31 @@ Example output (pretty-printed)::
 
 Secure Storage
 ==============
+Use the Secure Storage HTTP RESTful API to create, retrieve, and delete secure keys.
+Details about secure storage and secure keys in CDAP can be found in :ref:`Administration
+Manual: Secure Storage <admin-secure-storage>`.
 
-Use the Secure Storage HTTP RESTful API to create, retrieve and delete secure keys. Details about
-secure storage in CDAP can be found at :ref:`Admin' Manual: Secure Storage <admin-secure-storage>`.
+**Note:** In CDAP 3.5.0, encryption and decryption of the contents only happens at the
+secure store, not at the client endpoints or while the data is transitting to the secure
+store. A later version of CDAP will provide security at both endpoints and while in
+transit.
 
 .. _http-restful-api-security-secure-storage-add:
 
 Add a Secure Key
 ----------------
-You can add a key to secure storage by making an HTTP PUT request to the URL::
+You can add a secure key to secure storage by making an HTTP PUT request to the URL::
 
   PUT /v3/namespaces/<namespace-id>/securekeys/<secure-key-id>
 
-with JSON-formatted body that contains the description of the key, the data to be stored at the key and a map of
-properties associated with the key::
+with a JSON-formatted body that contains the description of the key, the data to be stored
+under the key, and a map of metadata properties associated with the key::
```

Comment:
so I steered clear of the word ``metadata`` to not have it be confused with the CDAP metadata system.